lib/logger_json.ex

```diff
@@ -59,7 +59,7 @@ defmodule LoggerJSON do
   """
   @behaviour :gen_event
 
-  @ignored_metadata_keys ~w[ansi_color initial_call crash_reason]a
+  @ignored_metadata_keys ~w[ansi_color initial_call crash_reason pid gl mfa report_cb]a
```

Comment:
I _think_ that this should cover all the [Erlang](http://erlang.org/doc/man/logger.html#type-metadata) and [Elixir](https://hexdocs.pm/logger/Logger.html#module-metadata) metadata keys.

Reply:
This is correct! Although a more fail proof approach is probably to traverse the metadata and reject anything that is a tuple, pid, function, or reference, as those cannot be JSON encoded. Something like:

    Enum.reject(metadata, fn {_, v} -> is_pid(v) or is_reference(v) or is_tuple(v) or is_function(v) end)

This means as Erlang/Elixir adds more options, this backend will continue working.

Reply:
@josevalim We would have to do this recursively right? This approach wouldn't catch a list of tuples for example.

Reply:
Good point. And things like structs is hard to know if they would be allowed or not. You would need to check the JSON protocol implementation. So this PR sounds great as a starting point after all.